components/viewer/ViewMenuBar.vue

```diff
@@ -23,9 +23,20 @@
 
 <script setup lang="ts">
 import ViewScoreStatus from '~/components/viewer/ViewScoreStatus.vue';
-import { useViewerStore } from '~/composables/store/viewer';
+import { useViewerRefs, useViewerStore } from '~/composables/store/viewer';
 
 const { toggleBackpack, toggleSearch } = useViewerStore();
+const { isSearchVisible } = useViewerRefs();
+
+const toggleSearchWrapper = (show: boolean) => {
+  toggleSearch(show);
+
+  nextTick(() => {
+    if (isSearchVisible.value) {
+      document.getElementById('search-input')?.focus();
+    }
+  });
```

Comment:
This code should probably be in the search modal itself.

You can use a [watcher](https://vuejs.org/guide/essentials/watchers) to monitor the visibility status and trigger the focus change. You should also use a [template ref](https://vuejs.org/guide/essentials/template-refs.html) to address elements in the DOM rather than use `document.getElementById()`.